src/executor/iExecutor.py

```diff
@@ -6,6 +6,12 @@ class iExecutor(ABC):
     def __init__(self, next: 'iExecutor'):
         self.next = next
 
+        self.database = None
+        super().__init__()
+
+    def set_database(self, database):
```

Comment:
not a huge deal, but I think the set_database may be better in a subclass since only the database executors should be actually using the database

Reply:
Good point, I made that change before our discussion, I'll move it back.